src/main/java/de/mari_023/fabric/ae2wtlib/util/ae2wtlibSubScreen.java

```diff
@@ -30,15 +31,18 @@
      */
     public ae2wtlibSubScreen(AEBaseScreen<?> gui, Object containerTarget) {
         this.gui = gui;
-        if(containerTarget instanceof WCTGuiObject) {//TODO don't hardcode
-            previousContainerIcon = new ItemStack(ae2wtlib.CRAFTING_TERMINAL);
-            previousContainerType = WCTContainer.TYPE;
-        } else if(containerTarget instanceof WPTGuiObject) {
-            previousContainerIcon = new ItemStack(ae2wtlib.PATTERN_TERMINAL);
-            previousContainerType = WPTContainer.TYPE;
+
+        if (containerTarget instanceof WTGuiObject) {
+            previousContainerType = ((WTGuiObject) containerTarget).type;
+            if (previousContainerType == WCTContainer.TYPE)
+                previousContainerIcon =  new ItemStack(ae2wtlib.CRAFTING_TERMINAL);
+            else if (previousContainerType == WPTContainer.TYPE)
+                previousContainerIcon = new ItemStack(ae2wtlib.PATTERN_TERMINAL);
+            else
```

Comment:
Actually, the best way might be to have a method like getIcon() in WTGuiObject and call that